src/Dstarlite.js

```diff
@@ -0,0 +1,115 @@
+import Heap from './Heap';
+import Cell from './Cell';
+
+export default class Dstarlite {
+    constructor(start, goal, heuristic) {
+        this.Heap = new Heap();
+        this.km = 0;
+        this.rhs = new Map();
+        this.gscore = new Map();
+        this.heuristic = heuristic;
+        this.start = start;
+        this.goal = goal;
+    }
+
+    initialize() {
+        const grid = this.goal.grid.grid;
+        for (const row of grid) {
+            for (const column of row) {
+                this.gscore.set(column, Infinity);
+                this.rhs.set(column, Infinity);
+            }
+        }
+
+        this.rhs.set(this.goal, 0);
+        this.Heap.insert(this.goal, this.calculateKey(this.goal));
+    }
+
+    calculateKey(node) {
+        const min = Math.min(this.gscore.get(node), this.rhs.get(node));
+        return [
+            min + this.heuristic(this.start, node) + this.km,
+            min,
+        ];
+    }
+
+    compareKeys(key1, key2) {
+        if (key1[0] > key2[0] || (key1[0] === key2[0] && key1[1] > key2[1])) {
+            return 1;
+        }
+        if (key1[0] === key2[0] && key1[1] === key2[1]) {
+            return 0;
+        }
+
+        return -1;
+    }
+
+    updateVertex(u) {
+        if (u !== this.goal) {
+            let rhs = Infinity;
+            for (const {node: successor} of u.getNeighbors(true)) {
+                rhs = Math.min(rhs, Cell.calculateCost(u, successor) + this.gscore.get(successor));
+            }
+            this.rhs.set(u, rhs);
+        }
+
+        this.Heap.remove(u);
+
+        if (this.gscore.get(u) !== this.rhs.get(u)) {
+            this.Heap.insert(u, this.calculateKey(u));
+        }
+    }
+
+    computeShortestPath() {
+        while (this.compareKeys(this.Heap.topKey(), this.calculateKey(this.start)) < 0 || this.rhs.get(this.start) !== this.gscore.get(this.start)) {
```

Comment:
141 chars (+8 indent) is pretty not-great for a single line 😬 

there's two thinks i think we could do here:
1) make an iteration counter and the while loop test puts an upper limit on it
2) then you could check compareKeys / rhs != g and `break` if both fail

that would flatten this out quite nicely probs

Reply:
also you're *still* missing line 25' 🤔

Reply:
🐇